### PR TITLE
Filter out non office hours events in ical parser

### LINKED ISSUE #282: Filter out "Lecture" calendar events
Prevent "Lecture" calendar events from creating a new office hour queue when parsed. For an example of what I mean, check out the Fundies 1 calendar on prod.

If blocked, contact @liustanley

### PR DESCRIPTION
**Pull request by willstenzel**
Closes #282

### CODE REVIEW

**Comment by willstenzel** [comment]
@dajinchu do you know how I can modify the `parsedICS` so I can test that it filters out non office hour events?

https://github.com/sandboxnu/office-hours/blob/4eba60a58a0da835051a07e4f408c04fffec300f/packages/server/src/course/ical.service.spec.ts#L12

File: packages/server/src/course/ical.service.ts [hunk 1]
```diff
@@ -27,7 +27,17 @@ export class IcalService {
         iCalElement.end !== undefined,
     );
 
-    return officeHours.map(event => ({
+    const isOfficeHoursEvent = (title: string) => {
+      const nonOfficeHourKeywords = ['Lecture', 'Lab', 'Exam', 'Class']
+      nonOfficeHourKeywords.forEach(keyword => {
```
Comment (NEUDitao) [inline packages/server/src/course/ical.service.ts:32]: andmap reduces line 34 and line 37 omo

**Review by NEUDitao — APPROVED** [review]
lgtm, to test it, edit the database entry for the course to have the right ical link and do `yarn cli ical:scrape`

File: packages/server/src/course/ical.service.ts (hunk 1 above)
Comment (NEUDitao) [reply packages/server/src/course/ical.service.ts:32]: `return nonOfficeHourKeywords.every(keyword => !title.includes(keyword));`

Comment (willstenzel) [reply packages/server/src/course/ical.service.ts]: oh sweet, ty!

**Pull request merged**